### PR TITLE
[SS-2017-004] FIX DataDifferencer doesnt correctly cast data

### PR DESCRIPTION
**Pull request by tractorcow**
Merge up from 3 branch for https://github.com/silverstripe/silverstripe-framework/issues/7025

### CODE REVIEW

**Comment by tractorcow** [comment]
Updated.

**Comment by tractorcow** [comment]
Fixed @flamerohr

**Comment by tractorcow** [comment]
Darn it, I just realised I was pushing to `versioned` but running the `silverstripe-assets` tests locally. =(

**Comment by tractorcow** [comment]
ok, now with being tested locally.

File: tests/php/DataDifferencerTest.php [hunk 1]
```diff
@@ -59,17 +74,51 @@ public function testArrayValues()
...
         // TODO Using getter would split up field again, bug only caused by simulating
         // an array-based value in the first place.
-        $this->assertContains('<ins>a</ins><del>a,b</del>', str_replace(' ', '', $obj1Diff->getField('Choices')));
+        $this->assertContainsIgnoreWhitespace('<ins>a</ins><del>a,b</del>', $obj1Diff->getField('Choices'));
+    }
+
+    public function testCast()
+    {
+        /** @var DataDifferencerTest\TestObject $obj1 */
+        $obj1 = $this->objFromFixture(DataDifferencerTest\TestObject::class, 'obj1');
```
Comment (flamerohr) [inline tests/php/DataDifferencerTest.php]: missing `use DataDifferencerTest\TestObject;`? ;)

I don't think that's a blocker, maybe if you feel like cleaning it up the next time you're in DataDifferencer.

Comment (tractorcow) [inline tests/php/DataDifferencerTest.php]: Yeah it's a bit weird in some places. We have quite a few TestObject classes though; probably should name them better.

**Pull request merged**